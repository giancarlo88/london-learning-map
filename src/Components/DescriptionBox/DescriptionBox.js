import React from 'react'
import PropTypes from 'prop-types'
import styled from '../../../node_modules/styled-components';
import CloseButton from '../CloseButton/CloseButton';


const StyledDescriptionBox = styled.div`
  background-color: white;
  position: fixed;
  right: 10px; 
  left: 10px;
  bottom: 10px;
  height: 30vh;

  @media (min-width: 480px) {
    bottom: initial; 
    left: initial;
    height: 75vh;
    top: 10px;
    width: 45%
  }

  @media (min-width: 768px) { 
    width: 30%;
  }
`

const DescriptionBox = ({description, title, handleCloseClick, index}) => (
  <StyledDescriptionBox>
    <CloseButton handleCloseClick={() => handleCloseClick(index)}/>
    <h1>{title}</h1>
    <p>{description}</p>
  </StyledDescriptionBox>
)

DescriptionBox.propTypes = { 
  description: PropTypes.string.isRequired,
  handleCloseClick: PropTypes.func.isRequired
}

export default DescriptionBox