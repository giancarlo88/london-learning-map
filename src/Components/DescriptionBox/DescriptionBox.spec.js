import React from 'react'
import { shallow } from 'enzyme'

import DescriptionBox from './DescriptionBox'

describe(DescriptionBox.name, () => {
  const description = 'here is a description'
  const props = {
    description, 
    handleCloseClick: () => {}
  }

  describe('@renders', () => {
    it('should render', () => {
      const wrapper = shallow(<DescriptionBox {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('@props', () => {
    it('should render prop text as text', () => {
      const wrapper = shallow(<DescriptionBox {...props} />)
      expect(wrapper.dive().find('p').text()).toBe(description)
    })
  })
})
