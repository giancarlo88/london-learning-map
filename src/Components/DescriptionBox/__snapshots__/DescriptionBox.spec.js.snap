// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionBox @renders should render 1`] = `
<styled.div>
  here is a description
</styled.div>
`;
